sound/soc/sof/pcm.c

```diff
@@ -470,6 +470,10 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		if (ret < 0)
 			return ret;
 
+		ret = snd_sof_pcm_platform_hw_free(sdev, substream);
+		if (ret < 0)
+			return ret;
+
```

Comment:
Sorry, I still don't get this. This would be called for start as well.

Reply:
@kv2019i, the `reset_hw_params` is only set to true in case of SNDRV_PCM_TRIGGER_SUSPEND or SNDRV_PCM_TRIGGER_STOP.
At line 468 we have the `if (!ret && reset_hw_params)`, so this part of the code is not run at start.

Reply:
How this is gong to affect non Intel platforms? Is there a possibility that they will break?

Reply:
Uh. @ujfalusi . I blame The "Firefox Github" problem. The indendation is all wrong and I saw this "hw_free" as a top-level statement at the end of function.

Reply:
@kv2019i, I had to open the file. Or copy paste a selection to a text file to see the correct indentation.
Still, I refuse to not use Firefox...

Reply:
> How this is gong to affect non Intel platforms? Is there a possibility that they will break?

@ujfalusi in theory this should not break non-Intel platforms because in both cases (stop and suspend), we restart the stream from the prepare stage which will call the platform_hw_params(). So, this would balance the calls. @ajitkupandey and @dbaluta, could you please confirm?